ir/src/symbol_table.rs

```diff
@@ -43,6 +76,12 @@ pub(super) struct SymbolTable {
 }
 
 impl SymbolTable {
+    /// Sets the number of trace segments, based on the current number of trace segments and the
+    /// provided trace segment identifier, which is indexed from zero.
```

Comment:
minor nit: Maybe this can be rephrased a bit to clarify that num_trace_segments is set to max of the trace segment encountered till now but it's easy to look at the code and see what's happening at a glance so maybe not required.

Reply:
I've attempted to clarify.